File: tests/go/cases/case_188_local_cache_named_retained_09.go
package escape_tests

var retainedCase188 = []map[string]string{}

func Case188LocalCacheNamedRetained09(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "local_cache_named_retained_09",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "local_cache_named_retained_09:" + raw,
	}
	retainedLocal := []map[string]string{}
	retainedLocal = append(retainedLocal, payload)
	// SAFE: local slice dies at return.
	return payload["checksum"]
}

File: tests/nodejs/cases/case_188_local_cache_named_retained_09.js
/**
 * Task case 188: local_cache_named_retained_09 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case188LocalCacheNamedRetained09(input) {
  const taskName = 'local_cache_named_retained_09';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const retainedCacheLocal = new Map();
  retainedCacheLocal.set('tmp', payload);
  // SAFE: local map does not escape.
  return payload.checksum;
}

module.exports = {
  case188LocalCacheNamedRetained09,
};
